build_projects/shared-build-targets-utils/Utils/Crossgen.cs

```diff
@@ -31,41 +31,82 @@ public Crossgen(string coreClrVersion)
 
         private string GetCrossgenPathForVersion()
         {
-            string arch = RuntimeEnvironment.RuntimeArchitecture;
-            string packageId;
+            var crossgenPackagePath = GetCrossGenPackagePathForVersion();
+
+            if (crossgenPackagePath == null)
+            {
+                return null;
+            }
+
+            return Path.Combine(
+                crossgenPackagePath,
+                "tools",
+                $"crossgen{Constants.ExeSuffix}");
+        }
+
+        private string GetLibCLRJitPathForVersion()
+        {
+            var coreclrRid = GetCoreCLRRid();
+            var crossgenPackagePath = GetCrossGenPackagePathForVersion();
+
+            if (crossgenPackagePath == null)
+            {
+                return null;
+            }
+
+            return Path.Combine(
+                crossgenPackagePath,
+                "runtimes",
+                coreclrRid,
+                "native",
+                $"{Constants.DynamicLibPrefix}clrjit{Constants.DynamicLibSuffix}");
+        }        
+
+        private string GetCrossGenPackagePathForVersion()
+        {
+            string coreclrRid = GetCoreCLRRid();
+
+            if (coreclrRid == null)
+            {
+                return null;
+            }
+
+            string packageId = $"runtime.{coreclrRid}.Microsoft.NETCore.Runtime.CoreCLR";            
+
+            return Path.Combine(
+                Dirs.NuGetPackages,
+                packageId,
+                _coreClrVersion);
+        }
+
+        private string GetCoreCLRRid()
```

Comment:
I was trying to find it, but I wasn't sure where it was and didn't want to get something that would return to me a RID that was similar but not the same. If someone can point me with certainty where this method is, I would LOVE to use it.